src/main/java/training/subsystems/Roller/RollerSubsystem.java

```diff
@@ -0,0 +1,67 @@
+package training.subsystems.Roller;
+
+import com.revrobotics.CANSparkBase;
+import com.revrobotics.CANSparkLowLevel;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkPIDController;
+import utils.GBSubsystem;
+
+import static training.subsystems.Roller.Constants.*;
+
+public class RollerSubsystem extends GBSubsystem {
+    private static RollerSubsystem instance;
+    private final CANSparkMax motor = new CANSparkMax(ROLLER_ID, CANSparkLowLevel.MotorType.kBrushless);
+    private final SparkPIDController motorPID = motor.getPIDController();
+    private double velocity;
+    private boolean run;
+
+    private RollerSubsystem() {
+        motorPID.setP(ROLLER_P);
+        motorPID.setI(ROLLER_I);
+        motorPID.setD(ROLLER_D);
+        motorPID.setOutputRange(0, POWER_LIMIT_ROLLER);
+        motor.burnFlash(); // applies some of the changes above
```

Comment:
r u sure it will not delete the pid also? but good practice is to flush before youre setting  anything

Reply:
According to the docs, I have to run burnflush to apply some settings, e.g. output range of the PID. From my understanding it just make sure some settings take effect, though I may be wrong, as the docs aren't very detailed about that.

Reply:
I would check about it again, I ask not really sure about the burn flush

Reply:
From the official docs:

> `public [REVLibError](https://codedocs.revrobotics.com/java/com/revrobotics/revliberror) setD​(double gain)`
Set the Derivative Gain constant of the PIDF controller on the SPARK. This uses the Set Parameter API and should be used infrequently. The parameter does not presist unless burnFlash() is called. The recommended method to configure this parameter is to use the REV Hardware Client to tune and save parameters.

Hence, burnFlash() is needed to set the D value (and btw, same for other methods). Also, under setRefrence is written:

> [...] value - The value to set depending on the control mode. For basic duty cycle control this should be a value between -1 and 1 Otherwise: Voltage Control: Voltage (volts) Velocity Control: Velocity (RPM) Position Control: Position (Rotations) Current Control: Current (Amps). Native units can be changed using the setPositionConversionFactor() or setVelocityConversionFactor() methods of the CANEncoder class

Since I use Position, Rotations is the specified as the right option.

burnFlash is described as:

> `public [REVLibError](https://codedocs.revrobotics.com/java/com/revrobotics/revliberror) burnFlash()`
Writes all settings to flash.
not that much. I don't know what's that flash is, but I don't see any reason for it to delete the PID -- which explicitly says that it's required for it to write the changes.